charts/ocis/templates/_common/_tplvalues.tpl

```diff
@@ -249,3 +249,45 @@ oCIS serviceAccount settings
 {{- define "ocis.serviceAccount" -}}
 automountServiceAccountToken: true
 {{- end -}}
+
+{{/*
+oCIS secret wrapper
+
+@param .name          The name of the secret.
+@param .params        Dict containing data keys/values (plaintext).
+@param .scope         The current scope
+*/}}
+{{- define "ocis.secret" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .name }}
+data:
+  {{- $secretObj := (lookup "v1" "Secret" .scope.Release.Namespace .name) | default dict }}
+  {{- $secretData := (get $secretObj "data") | default dict }}
+  {{- range $key, $value := .params }}
+  {{- $secretValue := (get $secretData $key) | default ($value | b64enc | quote)}}
+  {{ $key }}: {{ $secretValue }}
+  {{- end }}
+{{- end -}}
+
+{{/*
+oCIS secret wrapper
+
+@param .name          The name of the secret.
+@param .params        Dict containing data keys/values (plaintext).
+@param .scope         The current scope
+*/}}
+{{- define "ocis.configMap" -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .name }}
+data:
+  {{- $configObj := (lookup "v1" "ConfigMap" .scope.Release.Namespace .name) | default dict }}
+  {{- $configData := (get $configObj "data") | default dict }}
+  {{- range $key, $value := .params }}
+  {{- $configValue := (get $configData $key) | default ($value | quote)}}
```

Comment:
You're right, I assumed (potentially wrongly) that the data of the ConfigMap would be properly quoted anwyay but now that I think of it, it can't hurt to pipe it to quote.